.github/workflows/ci.yml

```
@@ -135,6 +136,31 @@ jobs:
           salt-version: ""
           validate-version: true
 
+      - name: Get Pull Number
+        id: get-pull-number
+        uses: ./.github/actions/get-pull-number
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          owner: ${{ github.repository_owner }}
+          repo: ${{ github.event.repository.name }}
+          sha: ${{ github.sha }}
+          pull-number: ${{ github.event.pull_request.number }}
+
+      - name: Get Pull Request
+        id: get-pull-request
+        uses: ./.github/actions/get-pull-request
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          owner: ${{ github.repository_owner }}
+          repo: ${{ github.event.repository.name }}
+          pull-number: ${{ steps.get-pull-number.outputs.number }}
+
+      - name: Get Pull Labels
+        id: get-pull-labels
+        uses: ./.github/actions/get-pull-labels
+        with:
+          pull-request: ${{ steps.get-pull-request.outputs.pull-request }}
```

```suggestion
      - name: Get Pull Number
        id: get-pull-number
        uses: ./.github/actions/get-pull-number
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          owner: ${{ github.repository_owner }}
          repo: ${{ github.event.repository.name }}
          sha: ${{ github.sha }}
          pull-number: ${{ github.event.pull_request.number }}

      - name: Get Pull Request
        id: get-pull-request
        uses: ./.github/actions/get-pull-request
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          owner: ${{ github.repository_owner }}
          repo: ${{ github.event.repository.name }}
          pull-number: ${{ steps.get-pull-number.outputs.number }}

      - name: Get Pull Labels
        id: get-pull-labels
        uses: ./.github/actions/get-pull-labels
        with:
          pull-request: ${{ steps.get-pull-request.outputs.pull-request }}
```
Why didn't you implement all this in python under `tools/`, more specifically `tools/ci.py`?

---

Because it felt the easiest and get pull number uses `gh` and I wanted them to all be kept in the same place